ccm_web/server/src/api/api.admin.handler.ts

```diff
@@ -102,8 +119,12 @@ export class AdminApiHandler {
     const result = makeResponse<CanvasCourse[]>(coursesResponses)
     if (isAPIErrorData(result)) return result
     const allCourses: CanvasCourse[] = []
-    result.map(cs => allCourses.push(...cs))
+    result.forEach(cs => allCourses.push(...cs))
     logger.debug(`Number of courses matching search term: ${allCourses.length}`)
+    if (allCourses.length > this.maxSearchCourses) {
+      this.logTooManyCourseResultsWarning(queryParams)
+      throw new TooManyResultsError()
```

Comment:
Will this code ever hit? by looking at the code from L90-L93 above if there are too many courses then you are throwing an exception and it will exit from this admin handler and go to the api service. 

What is the use case this code is necessary? if I am not mistaken the function `getCourseSectionsInTerm` is only called when searching by course name.

Reply:
So, it's possible that the user will be a sub-account admin in two or more accounts and that while each individual account doesn't have more than 400 courses matching, all together they have greater than 400 courses. This is checking whether that is the case for consistency. Make sense?

Reply:
Ok, got it make sense now.